crates/bvs-vault-cw20-tokenized/src/contract.rs

```diff
@@ -0,0 +1,705 @@
+    pub fn balance_of(deps: Deps, staker: String) -> StdResult<Uint128> {
+        // this func come from the cw20_base crate
+        // validate the staker address
+        let balance = query_balance(deps, staker)?;
+
+        StdResult::Ok(balance.balance)
+    }
+
+    /// Get the staking token of staker, converted from receipt_tokens held by staker.
+    pub fn assets(deps: Deps, env: Env, staker: Addr) -> StdResult<Uint128> {
+        let balance = query_balance(deps, staker.to_string())?;
+        convert_to_staking_token(deps, env, balance.balance)
+    }
+
+    /// Given the number of receipt_token, convert to staking token based on the vault exchange rate.
+    pub fn convert_to_staking_token(
+        deps: Deps,
+        env: Env,
+        receipt_tokens: Uint128,
+    ) -> StdResult<Uint128> {
+        let underlying_token_balance = StakingToken::query_balance(&deps, &env)?;
+        let receipt_token_supply = cw20_base::contract::query_token_info(deps)?.total_supply;
+        let vault = offset::VirtualOffset::new(receipt_token_supply, underlying_token_balance)?;
+        vault.shares_to_assets(receipt_tokens)
+    }
+
+    /// Given assets, get the resulting receipt tokens based on the vault exchange rate.
+    /// Shares in this tokenized vault the receipt token.
+    /// Keeping the msg name the same as the non-tokenized vault for consistency.
+    pub fn convert_to_receipt_token(deps: Deps, env: Env, assets: Uint128) -> StdResult<Uint128> {
+        let underlying_token_balance = StakingToken::query_balance(&deps, &env)?;
+        let receipt_token_supply = cw20_base::contract::query_token_info(deps)?.total_supply;
+        let vault = offset::VirtualOffset::new(receipt_token_supply, underlying_token_balance)?;
+        vault.assets_to_shares(assets)
+    }
+
+    /// Total issued receipt tokens.
+    /// AKA total shares in the vault.
+    /// AKA Total circulating supply of the receipt token.
+    pub fn total_receipt_token_supply(deps: Deps, _env: Env) -> StdResult<Uint128> {
+        let receipt_token_supply = cw20_base::contract::query_token_info(deps)?.total_supply;
+        StdResult::Ok(receipt_token_supply)
+    }
+
+    /// Total Staking Tokens in this vault. Including assets through staking and donations.
+    pub fn total_assets(deps: Deps, env: Env) -> StdResult<Uint128> {
+        StakingToken::query_balance(&deps, &env)
+    }
+
+    /// Get the queued withdrawal info in this vault.
+    pub fn queued_withdrawal(deps: Deps, staker: Addr) -> StdResult<QueuedWithdrawalInfo> {
+        shares::get_queued_withdrawal_info(deps.storage, &staker)
+    }
+
+    /// Returns the vault information
+    pub fn vault_info(deps: Deps, env: Env) -> StdResult<VaultInfoResponse> {
+        let balance = StakingToken::query_balance(&deps, &env)?;
+        let receipt_token_supply = cw20_base::contract::query_token_info(deps)?.total_supply;
+        let cw20_contract = StakingToken::get_cw20_contract(deps.storage)?;
```

Comment:
```suggestion
        UnderlyingToken::query_balance(&deps, &env)
    }

    /// Get the queued withdrawal info in this vault.
    pub fn queued_withdrawal(deps: Deps, staker: Addr) -> StdResult<QueuedWithdrawalInfo> {
        shares::get_queued_withdrawal_info(deps.storage, &staker)
    }

    /// Returns the vault information
    pub fn vault_info(deps: Deps, env: Env) -> StdResult<VaultInfoResponse> {
        let balance = UnderlyingToken::query_balance(&deps, &env)?;
        let receipt_token_supply = cw20_base::contract::query_token_info(deps)?.total_supply;
        let cw20_contract = UnderlyingToken::get_cw20_contract(deps.storage)?;
```

For consistency sake.

Reply:
Not a fan of underlying token term for this vault. Will apply suggestion just out of consistency sake.

Reply:
Why not a fan? Would love to hear the reason